cool-core/src/main/java/com/nus/cool/core/util/writer/NativeDataWriter.java

```diff
@@ -170,7 +180,9 @@ public boolean add(FieldValue[] tuple) throws IOException {
     }
     // start a new chunk
     if (maybeSwitchChunk(curUser)) {
-      maybeSwitchCublet();
+      if (maybeSwitchCublet()) {
```

Comment:
It is better to create a new data chunk. Then we can keep the offset field of data chunk immutable and not exposing a setter.